File: ios/Classes/FlutterImageGallerySaverPlugin.swift
import Flutter
import UIKit

public class FlutterImageGallerySaverPlugin: NSObject, FlutterPlugin { 
  public static func register(with registrar: FlutterPluginRegistrar) {
    let messenger = registrar.messenger()
    let api = ImageGallerySaverApiImpl()
    ImageGallerySaverApiSetup.setUp(binaryMessenger: messenger, api: api)
  }
}

File: android/src/main/kotlin/dev/knottx/flutter_image_gallery_saver/FlutterImageGallerySaverPlugin.kt
package dev.knottx.flutter_image_gallery_saver

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
 
import dev.knottx.flutter_image_gallery_saver.ImageGallerySaverApi

class FlutterImageGallerySaverPlugin : FlutterPlugin {
    private lateinit var applicationContext: Context

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        applicationContext = binding.applicationContext
        ImageGallerySaverApi.setUp(
            binding.binaryMessenger,
            ImageGallerySaverApiImpl(applicationContext),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) { 
        ImageGallerySaverApi.setUp(binding.binaryMessenger, null)
    }
}